disttools/setup.py

```diff
@@ -1,15 +1,34 @@
+import os
 import pathlib
 import subprocess
 
 from setuptools import setup
+from setuptools.command.install import install
 from wheel.bdist_wheel import bdist_wheel
 
 base_dir = pathlib.Path(__file__).parent.resolve()
 
 
+class LegacyInstallUnsupported(Exception):
+    pass
+
+
+def check_pep517() -> None:
+    if "PEP517_BUILD_BACKEND" not in os.environ:
+        raise LegacyInstallUnsupported("Rerun pip with --use-pep517")
```

Comment:
What are the effects of this when the package is installed as a dependency of RecordFlux? Does RecordFlux also need to be installed with `--use-pep517`?

Reply:
Yes. I tried that with testpypi. When adopting 0.9.2 in RecordFlux, we'll have to update the instructions. Do you see any problems here?

Reply:
Right now all RecordFlux dependencies seem to be PEP 517 compliant. Yes, a PEP 517 install seems to be done automatically when all requirements are met. However, using `--use-pep517` is the only way that I found to avoid a fallback to a legacy install when the PEP 517 install fails (e.g. because you done have `gprbuild` in you path). The fallback is what causes the silent installation of the failed build. I'm open for suggestions, but the whole Python packaging stuff is not easy to understand (and building native Ada libraries is not exactly a standard procedure).

Reply:
> Yes, a PEP 517 install seems to be done automatically when all requirements are met.

That seems not to be the case for me. The installation fails if `--use-pep517` is not given:

```
% pip install build/RecordFlux-parser-0.9.2.tar.gz
Processing ./build/RecordFlux-parser-0.9.2.tar.gz
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: RecordFlux-parser
  Building wheel for RecordFlux-parser (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/devel/Componolit/RecordFlux-parser/venv/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-req-build-3jg58cg0/setup.py'"'"'; __file__='"'"'/tmp/pip-req-build-3jg58cg0/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-95xz0w7j
       cwd: /tmp/pip-req-build-3jg58cg0/
  Complete output (18 lines):
  running bdist_wheel
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/tmp/pip-req-build-3jg58cg0/setup.py", line 34, in <module>
      setup(
    File "/home/devel/Componolit/RecordFlux-parser/venv/lib/python3.9/site-packages/setuptools/__init__.py", line 153, in setup
      return distutils.core.setup(**attrs)
    File "/usr/lib/python3.9/distutils/core.py", line 148, in setup
      dist.run_commands()
    File "/usr/lib/python3.9/distutils/dist.py", line 966, in run_commands
      self.run_command(cmd)
    File "/usr/lib/python3.9/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/tmp/pip-req-build-3jg58cg0/setup.py", line 23, in run
      check_pep517()
    File "/tmp/pip-req-build-3jg58cg0/setup.py", line 18, in check_pep517
      raise LegacyInstallUnsupported("Rerun pip with --use-pep517")
  __main__.LegacyInstallUnsupported: Rerun pip with --use-pep517
  ----------------------------------------
  ERROR: Failed building wheel for RecordFlux-parser
  Running setup.py clean for RecordFlux-parser
Failed to build RecordFlux-parser
[...]
```

Reply:
You're right. The source package was missing a `pyproject.toml` to attempt a PEP 517 compatible install automatically. Have a look at fa453c8.